### PR TITLE
Fix https state

### LINKED ISSUE #32: Update to account for removal of HTTPS state
HTTPS state has been removed and this document does not account for that. I'm also not entirely sure it defines the right logic for workers. In particular data: URL workers might need some more consideration.

If this specification is being maintained I can look into writing a PR, but the open PR does not seem reassuring.

cc @domenic

### PR DESCRIPTION
**Pull request by carlosjoan91**
Removes all mentions of HTTPS state in MIX and MIX2.

Also fixes a typo ('requires prohibits') and link issues in MIX2

Fixes #32

***
<a href="https://pr-preview.s3.amazonaws.com/carlosjoan91/webappsec-mixed-content/pull/33.html" title="Last updated on Sep 11, 2020, 6:37 PM UTC (450341c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webappsec-mixed-content/33/aed23ee...carlosjoan91:450341c.html" title="Last updated on Sep 11, 2020, 6:37 PM UTC (450341c)">Diff</a>

### CODE REVIEW

File: index.src.html [hunk 1]
```diff
@@ -494,8 +489,10 @@ <h3 id="categorize-settings-object">
             2.  Let |embedder settings| be |document|'s [=global object=]'s
                 [=relevant settings object=].
 
-            3.  If |embedder settings|' [=environment settings object/HTTPS state=] is not "`None`",
-                then return "`Prohibits Mixed Security Contexts`".
+            3.  If |embedder settings|'
+                [=environment settings object/origin=] is <a><i lang="la">
+                a priori</i> authenticated</a>., then return
```
Comment (annevk) [inline index.src.html:494]:
```suggestion
                a priori</i> authenticated</a>, then return
```

File: level2.src.html [hunk 2]
```diff
@@ -361,8 +356,10 @@ <h3 id="categorize-settings-object">
             2.  Let |embedder settings| be |document|'s [=global object=]'s
                 [=relevant settings object=].
 
-            3.  If |embedder settings|' [=environment settings object/HTTPS state=] is not "`None`",
-                then return "`Prohibits Mixed Security Contexts`".
+            3.  If |embedder settings|'
+                [=environment settings object/origin=] is <a><i lang="la">
+                a priori</i> authenticated</a>., then return
```
Comment (annevk) [inline level2.src.html:361]:
```suggestion
                a priori</i> authenticated</a>, then return
```

**Review by annevk — APPROVED** [review]
Thanks, looks good modulo nits.

I'm pretty sure the entirety of the WebSocket section can go, but there's also a separate issue tracking that so it can be done later.

**Comment by carlosjoan91** [comment]
Thanks for reviewing! Re: Websockets, yeah, I agree that the section can indeed be removed, I'll address that in a separate PR.

**Review by tittyboi — COMMENTED** [review]
Y

**Pull request merged**